spec/javascripts/components/details-spec.js

```diff
@@ -76,4 +76,17 @@ describe('Details component', function () {
 
     expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith('track-category', 'track-action', { label: 'closed' })
   })
+
+  it('allows custom track options', function () {
+    $('.gem-c-details').attr('data-track-action', 'track-action')
+    $('.gem-c-details').attr('data-track-category', 'track-category')
+    $('.gem-c-details').attr('data-track-options', '{"value":"track-value"}')
```

Comment:
Is it worth making this more complicated or is this an adequate test? I don't know, I'm genuinely asking.

Reply:
I think it's adequate for now as the main thing was to check that it converts it properly and passes it in